Build fails on macOS:

```
building '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv'...
unpacking sources
unpacking source archive /nix/store/wlwlah3xxfhwbm76sjzdm0r5hf29rk53-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/xxfxh4npdlp0w5winp2f17ji1yxdivn3-bash-4.4-p23/bin/bash
gcc -Wall -Wextra -g -Werror=missing-declarations -Werror=redundant-decls -Iinclude -c src/draw.c -o src/draw.o
/nix/store/xxfxh4npdlp0w5winp2f17ji1yxdivn3-bash-4.4-p23/bin/bash: gcc: command not found
make: *** [Makefile:41: src/draw.o] Error 127
builder for '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv' failed with exit code 2
error: build of '/nix/store/4ysqcig9hv69z232hslhw0cxjhsk4ljg-programmer-calculator-2.1.drv' failed
```

Adding `ghc` to the `buildInputs` fixes it for me.

--- a/pkgs/applications/science/math/programmer-calculator/default.nix
+++ b/pkgs/applications/science/math/programmer-calculator/default.nix
@@ -1,0 +1,34 @@
+{ lib, gccStdenv, fetchFromGitHub, ncurses }:
+
+gccStdenv.mkDerivation rec {
+  pname = "programmer-calculator";
+  version = "2.1";
+
+  src = fetchFromGitHub {
+    owner = "alt-romes";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1vvpbj24ijl9ma0h669n9x0z1im3vqdf8zf2li0xf5h97b14gmv0";
+  };
+
+  buildInputs = [ ncurses ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm 555 pcalc -t "$out/bin"
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A terminal calculator for programmers";
+    longDescription = ''
+      Terminal calculator made for programmers working with multiple number
+      representations, sizes, and overall close to the bits
+    '';
+    homepage = "https://alt-romes.github.io/programmer-calculator";
+    changelog = "https://github.com/alt-romes/programmer-calculator/releases/tag/v${version}";
+    license = licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ cjab ];
+    platforms = platforms.all;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -30173,6 +30173,8 @@ in
 
   pcalc = callPackage ../applications/science/math/pcalc { };
 
+  programmer-calculator = callPackage ../applications/science/math/programmer-calculator { };
+
   bcal = callPackage ../applications/science/math/bcal { };
 
   pspp = callPackage ../applications/science/math/pspp { };

